src/main/java/nl/rijksoverheid/mev/gezagsmodule/domain/gezagvraag/OuderOverledenOfOnbevoegdTotGezag.java

```diff
@@ -28,36 +42,43 @@ protected OuderOverledenOfOnbevoegdTotGezag(final GezagsBepaling gezagsBepaling)
     @Override
     public void perform() {
         Persoonslijst plPersoon = gezagsBepaling.getPlPersoon();
-        answer = V4A_3_JA;
-        
-        // Preconditie: minimaal 1 ouder moet een BSN hebben
-        Ouder1 persoonOuder1 = plPersoon.getOuder1();
-        Ouder2 persoonOuder2 = plPersoon.getOuder2();
-        if ((persoonOuder1 == null || persoonOuder1.getBurgerservicenummer() == null)
-            && (persoonOuder2 == null || persoonOuder2.getBurgerservicenummer() == null)) {
+        var heeftOuder1Burgerservicenummer = plPersoon.getOuder1AsOptional().map(Ouder1::getBurgerservicenummer).isPresent();
+        var heeftOuder2Burgerservicenummer = plPersoon.getOuder2AsOptional().map(Ouder2::getBurgerservicenummer).isPresent();
+        if (!heeftOuder1Burgerservicenummer && !heeftOuder2Burgerservicenummer) {
             throw new AfleidingsregelException("Preconditie: Ouder moet een BSN hebben", "Ouder moet een BSN hebben");
         }
-        Persoonslijst lplOuder1 = gezagsBepaling.getPlOuder1();
-        Persoonslijst lplOuder2 = gezagsBepaling.getPlOuder2();
-        if ((lplOuder1 == null) && (lplOuder2 == null)) {
+
+        Persoonslijst persoonslijstOuder1 = gezagsBepaling.getPlOuder1();
+        Persoonslijst persoonslijstOuder2 = gezagsBepaling.getPlOuder2();
+        if ((persoonslijstOuder1 == null) && (persoonslijstOuder2 == null)) {
             throw new AfleidingsregelException("Preconditie: Ouder moet geregistreerd staan in het BRP", "minimaal 1 ouder van de bevraagde persoon moet geregistreerd staan in het BRP");
         }
-        if (lplOuder1 != null
-            && (!lplOuder1.isOverledenOfOnbevoegd())) {
+
+        if (persoonslijstOuder1 != null && !persoonslijstOuder1.isOverledenOfOnbevoegd()) {
             answer = V4A_3_NEE_OUDER1;
         }
-        if (lplOuder2 != null
-            && (!lplOuder2.isOverledenOfOnbevoegd())) {
-            if (answer.equals(V4A_3_NEE_OUDER1)) {
+        if (persoonslijstOuder2 != null && !persoonslijstOuder2.isOverledenOfOnbevoegd()) {
+            if (V4A_3_NEE_OUDER1.equals(answer)) {
                 answer = V4A_3_NEE;
             } else {
                 answer = V4A_3_NEE_OUDER2;
             }
+        } else if (answer == null) { // een ouder is overleden of onbevoegd terwijl de ander onbekend is, of beide zijn overleden of onbevoegd
```

Comment:
sonar: comment verwijderen of opnemen in de javadoc

Reply:
Weet je zeker het een Sonar issue is? Sonar geeft bij mij 0 issues. Onze SonarQube Cloud integratie geeft ook geen nieuwe issues aan.

Inhoudelijk vind ik de comment verhelderend, omdat de reeks aan if/if-else statements verrassend verraderlijk is.

Reply:
Besproken.